easybuild/easyblocks/l/lammps.py

```diff
@@ -579,7 +581,21 @@ def get_kokkos_arch(kokkos_cpu_mapping, cuda_cc, kokkos_arch, cuda=None):
 
     processor_arch = None
 
-    if kokkos_arch:
+    if build_option('optarch') == OPTARCH_GENERIC:
+        # For generic Arm builds we use an existing target;
+        # this ensures that KOKKOS_ARCH_ARM_NEON is enabled (Neon is required for armv8-a).
+        # For other architectures we set a custom/non-existent type, which will disable all optimizations,
+        # and it should use the compiler (optimization) flags set by EasyBuild for this architecture.
+        if get_cpu_architecture() == systemtools.AARCH64:
+            processor_arch = 'ARMV80'
+        else:
+            processor_arch = 'EASYBUILD_GENERIC'
+
+        warning_msg = "Generic build requested, setting CPU ARCH to %s." % processor_arch
+        if kokkos_arch:
+            warning_msg += " The specified kokkos_arch (%s) will be ignored." % kokkos_arch
+        print_warning(warning_msg)
```

Comment:
Warnings should only be printed when something goes wrong or some fallback/override is happening, setting `CPU_ARCH` is fully intentional just a log message is enough there I think:

```suggestion
        self.log.info("Generic build requested, setting CPU ARCH to %s." % processor_arch)
        if kokkos_arch:
            msg = "The specified kokkos_arch (%s) will be ignored " % kokkos_arch
            msg += "because a generic build was requested (via --optarch=GENERIC)"
            print_warning(msg)
```